pkg/secrets/fetch_secret_test.go

```diff
@@ -144,6 +152,76 @@ func TestExecCommandError(t *testing.T) {
 	assert.Equal(t, "error while running './test/response_too_long/response_too_long"+binExtension+"': command output was too long: exceeded 20 bytes", err.Error())
 }
 
+func TestExecCommandWithHash(t *testing.T) {
+	prevCheckConfigFilePermissions := checkConfigFilePermissions
+	checkConfigFilePermissions = func(string) error { return nil }
```

Comment:
These operations could generate flaky test results if the tests are executed in parallel. You should migrate to local variables, and stop monkey-patching global instances.

Reply:
Thank you @Zenithar. Point taken, this was brought up in other comments, my plan is to change these, but in a separate PR.